### PR TITLE
Order manifests for shuttle orders. Now with item names and approver info.

### PR DESCRIPTION
**Pull request by Rinkashikachi**
## About the PR 
First of all creation of order manifests was implemented only for Telepads. With introduction of shuttles manifests basically disappeared and quartermasters had to write them by hand (if they care about bureaucracy ofc).

Next I've added approvers full name and job title to order manifests. This information is important to keep track of if you are a thorough quartermaster. Right now this feature is not that useful since manifests end up somewhere on the station together with the crate. But with introduction of copy machines in the future this will allow to get a ready to archive document with everything needed written in it.

For the same reason I've added the name of the ordered item itself. This is even more important than approver info, since order manifests were kinda useless anyway. They used to say just "order No1". Didn't make any sense to make copies of such documents or store them in the cargo office, since they were useless without the crate.

**Changelog**

:cl: Rinkashikachi
- add: added item name to order manifest
- add: added approvers full name and job title to order manifest
- add: added order manifest to shuttle orders

### CODE REVIEW

File: Content.Shared/Cargo/CargoOrderData.cs
Comment (metalgearsloth) [inline Content.Shared/Cargo/CargoOrderData.cs:27]: Why was the constructor order changed?

File: Content.Shared/Cargo/CargoOrderData.cs
Comment (metalgearsloth) [inline Content.Shared/Cargo/CargoOrderData.cs:18]: This should be 1 string, not 2.

File: Content.Shared/Cargo/CargoOrderData.cs
Comment (metalgearsloth) [inline Content.Shared/Cargo/CargoOrderData.cs:32]: As above.

File: Content.Server/Cargo/Systems/CargoSystem.Telepad.cs
Comment (metalgearsloth) [inline Content.Server/Cargo/Systems/CargoSystem.Telepad.cs:155]: .

File: Content.Shared/Cargo/CargoOrderData.cs
Comment (metalgearsloth) [inline Content.Shared/Cargo/CargoOrderData.cs:18]: You could probably also make the Approver a nullable string and then Approved just checks if it's not null.

File: Content.Server/Cargo/Systems/CargoSystem.Shuttle.cs
Comment (metalgearsloth) [inline Content.Server/Cargo/Systems/CargoSystem.Shuttle.cs:440]: C# style guide please (pascal case method names).

File: Content.Server/Cargo/Systems/CargoSystem.Shuttle.cs
Comment (metalgearsloth) [inline Content.Server/Cargo/Systems/CargoSystem.Shuttle.cs:470]: It's attaching the label to... the cargo shuttle?

File: Content.Server/Cargo/Systems/CargoSystem.Shuttle.cs
Comment (metalgearsloth) [inline Content.Server/Cargo/Systems/CargoSystem.Shuttle.cs:463]: .

**Review by metalgearsloth — DISMISSED** [review]

File: Content.Shared/Cargo/CargoOrderData.cs
Comment (Rinkashikachi) [reply Content.Shared/Cargo/CargoOrderData.cs:27]: I did this to reflect the new order of how these are written in the manifest.

File: Content.Server/Cargo/Systems/CargoSystem.Telepad.cs
Comment (Rinkashikachi) [reply Content.Server/Cargo/Systems/CargoSystem.Telepad.cs:155]: Will fix in the next commit.

File: Content.Server/Cargo/Systems/CargoSystem.Shuttle.cs
Comment (Rinkashikachi) [reply Content.Server/Cargo/Systems/CargoSystem.Shuttle.cs:440]: Will fix in the next commit. My bad, habit

File: Content.Server/Cargo/Systems/CargoSystem.Shuttle.cs
Comment (Rinkashikachi) [reply Content.Server/Cargo/Systems/CargoSystem.Shuttle.cs:463]: Will fix in the next commit.

File: Content.Server/Cargo/Systems/CargoSystem.Shuttle.cs
Comment (Rinkashikachi) [reply Content.Server/Cargo/Systems/CargoSystem.Shuttle.cs:470]: I'm still getting used to C# and the project itself. This piece of code is copied from CargoSystem.Telepad and I didn't understand why this is done like that as well. So I decided to leave it as it is (it worked).
Will fix in the next commit.

File: Content.Shared/Cargo/CargoOrderData.cs
Comment (Rinkashikachi) [reply Content.Shared/Cargo/CargoOrderData.cs:18]: Are you suggesting something like that?
```
public bool Approved;
{
    get { return Approver is null; }
}
```

Comment (metalgearsloth) [reply Content.Shared/Cargo/CargoOrderData.cs]: yeah, or just inline it like `public bool Approved => Approver != null;`

**Comment by metalgearsloth** [comment]
good first pr tho

**Comment by Rinkashikachi** [comment]
> good first pr tho

I didn't really work with GitHub before. Do I just press "Resolve conversation" after I've addressed it in the code or what?

**Comment by metalgearsloth** [comment]
> > good first pr tho
> 
> I didn't really work with GitHub before. Do I just press "Resolve conversation" after I've addressed it in the code or what?

yeah, then when they're all done you can request the re-review button.

**Comment by Rinkashikachi** [comment]
Not sure if I am supposed to do something else or you are just busy with other stuff. "Changes requested" confuses me

**Comment by Rinkashikachi** [comment]
> > > good first pr tho
> > 
> > 
> > I didn't really work with GitHub before. Do I just press "Resolve conversation" after I've addressed it in the code or what?
> 
> yeah, then when they're all done you can request the re-review button.

Not sure if I am supposed to do something else or you are just busy with other stuff. "Changes requested" confuses me

**Comment by Just-a-Unity-Dev** [comment]
> Not sure if I am supposed to do something else or you are just busy with other stuff. "Changes requested" confuses me

now you play the waiting game, maintainers have 10s of other PRs to review and check, so you'll probably have it up for a while until it get's merged

**Comment by Rinkashikachi** [comment]
> > Not sure if I am supposed to do something else or you are just busy with other stuff. "Changes requested" confuses me
> 
> now you play the waiting game, maintainers have 10s of other PRs to review and check, so you'll probably have it up for a while until it get's merged

yeah, don't have any problems with waiting. Just wanted to be sure, that I've addressed the request properly

File: Content.Shared/Cargo/CargoOrderData.cs
Comment (mirrorcult) [inline Content.Shared/Cargo/CargoOrderData.cs:19]:
```suggestion
        public bool Approved => Approver is not null;
```

File: Content.Shared/Cargo/CargoOrderData.cs
Comment (mirrorcult) [inline Content.Shared/Cargo/CargoOrderData.cs:31]:
```suggestion
        public void SetApproverData(IdCardComponent? idCard)
```
pascalcase

File: Content.Server/Cargo/Systems/CargoSystem.Orders.cs
Comment (mirrorcult) [inline Content.Server/Cargo/Systems/CargoSystem.Orders.cs:158]:
```suggestion
            order.SetApproverData(idCard);
```

**Review by mirrorcult — CHANGES_REQUESTED** [review]

**Review by mirrorcult — APPROVED** [review]

**Comment by Just-a-Unity-Dev** [comment]
> > > Not sure if I am supposed to do something else or you are just busy with other stuff. "Changes requested" confuses me
> > 
> > 
> > now you play the waiting game, maintainers have 10s of other PRs to review and check, so you'll probably have it up for a while until it get's merged
> 
> yeah, don't have any problems with waiting. Just wanted to be sure, that I've addressed the request properly

nice first PR though, looks really good

**Pull request merged**
